scripts/Muon/GUI/ElementalAnalysis2/auto_widget/ea_auto_tab_model.py

```diff
@@ -100,17 +99,18 @@ def _run_peak_algorithms(self, parameters):
         workspace = parameters["workspace"]
         run, detector = self.split_run_and_detector(workspace)
         if run is None or detector is None:
-            group = retrieve_ws(workspace)
-            for workspace_name in group.getNames():
-                if is_group_valid(workspace_name):
+            group_workspace = retrieve_ws(workspace)
+            for group in self.context.group_context.groups:
+                if group.run_number == workspace:
+                    workspace_name = group.get_counts_workspace_for_run(workspace, rebin=False)
```

Comment:
Can you talk me through your decision process for these few lines? I can understand renaming the variable from group to group_workspace but why remove the is_group_valid check and replace 2 lines of code with 3?

Reply:
Originally the code gets all the workspaces in the group workspace and then checks if it is valid as it could have rebinned workspaces or tables, while this way, it cycles through all groups in the group context then checks if it belongs to the correct group workspace, and then gets workspace name from group.

Reply:
Does this mean it will now include rebinned workspaces or tables? If so, do we want it to do that?

Reply:
I believe this is the best way to keep and access all information relating to each group, so whatever data that is needed can be easily retrieved from group context.